### ISSUE TITLE
Regression in 3.2.3 regarding implicit id

### ISSUE THREAD

**Issue by leopiccionia** [vuejs/vuefire#822]
### Reproduction

In `vuexfire@3.2.3`, bind a Firestore ref to a collection (e.g. `'mycollection'`) whose documents doesn't have an explicit `id` field.

### Steps to reproduce the behavior

Try to access the non-enumerable `id` field (= document key) inside your component, and notice it's `undefined`.

I noticed this issue when the following code started to throw because `doc` expected the first parameter to be non-empty string, but received `undefined` instead. It used to work in `vuexfire@3.2.2`.

```js
// doc is an array element in bound collection
firebase.collection('mycollection').doc(doc.id).update({ status: 'something' })
```

### Expected behavior

`doc` has non-enumerable property `id`, equal to document's key in database.

I'm not sure if [this](https://vuefire.vuejs.org/vuexfire/binding-subscriptions.html#key-id) makes it a documented feature.

### Actual behavior

`doc.id` is `undefined`.

### Additional information

In my tests, rolling back to `vuexfire@3.2.2` actually fixes the issue.

**Comment by posva** [comment]
Please provide a boiled down reproduction using one of the examples

**Comment by leopiccionia** [comment]
Upon further inspection, the issue seems to be related to a multi-tab synchronization library (non-enumerable keys weren't serialized). Changing version worked because I was clearing storage before my tests.

As the issue is unrelated to Vuexfire, I'm closing this. Sorry for the inconvenience!

**Comment by TheoWolf** [comment]
@leopiccionia I'm having the same issue, could you please give more insight on the multi-tab synchronization library and the version used to overcome this issue. Thanks

**Comment by leopiccionia** [comment]
@TheoWolf For now, I have just disabled the multi-tab synchronization, what degrades the experience.

I'll try to solve this soon (app's not yet in production), and let you know if I find something. But some hints that I have collected:
* Hook into post-hydration, like discussed [here](https://github.com/vuejs/vuefire/issues/805#issuecomment-647508458)
* Use [`serialize` option](https://vuefire.vuejs.org/api/vuexfire.html#firestoreaction), to make property enumerable, etc. (`JSON.stringify` ignores non-enumerable properties, by spec); we may have to remove that key before setting/updating the document
